packages/js/src/dashboard/components/google-site-kit-connection-widget.js

```diff
@@ -0,0 +1,158 @@
+import { ReactComponent as YoastConnectSiteKit } from "../../../images/yoast-connect-google-site-kit.svg";
+import { __ } from "@wordpress/i18n";
+import { CheckCircleIcon } from "@heroicons/react/solid";
+import { ArrowRightIcon, DotsVerticalIcon, XIcon, TrashIcon } from "@heroicons/react/outline";
+import { useSelect } from "@wordpress/data";
+import { addQueryArgs } from "@wordpress/url";
+import classNames from "classnames";
+
+/**
+ * Get the button and stepper props based on the current state.
+ *
+ * @param {boolean} active Whether the feature is active.
+ * @param {boolean} installed Whether the plugin is installed.
+ * @param {boolean} setup Whether the setup is complete.
+ * @param {boolean} connected Whether the connection is active.
+ * @param {string} installUrl The URL to install Site Kit.
+ * @param {string} activateUrl The URL to activate Site Kit.
+ * @param {string} setupUrl The URL to setup Site Kit.
+ *
+ * @returns {Object} The button and stepper props.
+ */
+const getButtonAndStepperProps = ( active, installed, setup, connected, installUrl, activateUrl, setupUrl ) => {
+	let buttonProps;
+	let currentStep;
+	let isComplete;
+
+	switch ( true ) {
+		case ( ! installed ):
+			currentStep = 1;
+			isComplete = false;
+			buttonProps = {
+				children: __( "Install Site Kit by Google", "wordpress-seo" ),
+				as: "a",
+				href: installUrl,
+			};
+			break;
+		case ( ! active ):
+			currentStep = 2;
+			buttonProps = {
+				children: __( "Activate Site Kit by Google", "wordpress-seo" ),
+				as: "a",
+				href: activateUrl,
+			};
+			break;
+		case ( ! setup ):
+			currentStep = 3;
+			buttonProps = {
+				children: __( "Set up Site Kit by Google", "wordpress-seo" ),
+				as: "a",
+				href: setupUrl,
+			};
+			break;
+		case ( ! connected ):
+			currentStep = 4;
+			buttonProps = { children: __( "Connect Site Kit by Google", "wordpress-seo" ) };
+			break;
+		case connected:
+			isComplete = true;
+			currentStep = 4;
+			buttonProps = { children: "Take a quick tour" };
+			break;
+	}
+	return { buttonProps, currentStep, isComplete };
+};
+
+/**
+ * The google site kit connection guide widget.
+ *
+ * @param {boolean} installed Whether the plugin is installed.
+ * @param {boolean} setup Whether the setup is complete.
+ * @param {boolean} active Whether the feature is active.
+ * @param {boolean} connected Whether the connection is active.
+ * @param {boolean} installed Whether the plugin is installed.
+ * @param {string} activateUrl The URL to activate Site Kit.
+ * @param {string} installUrl The URL to install Site Kit.
+ * @param {function} onRemove The function to call when the widget is removed.
+ * @param {function} onRemovePermanently The function to call when the widget is removed permanently.
+ *
+ * @returns {JSX.Element} The widget.
+ */
+export const GoogleSiteKitConnectionWidget = ( {
+	installUrl,
+	activateUrl,
+	setupUrl,
+	connected,
+	active,
+	setup,
+	installed,
```

Comment:
We usually indicate boolean names with `isX` to indicate it is a boolean true/false thing.